api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -1221,4 +1224,19 @@ export class ScenariosController {
       }
     }
   }
+
+  @Get(':scenarioId/editing-locks')
+  @ApiOperation({ summary: 'Get all locks' })
+  async findLocksForScenario(
+    @Param('scenarioId', ParseUUIDPipe) scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<ScenarioLockResultPlural> {
+    const result = await this.service.findAllLocks(scenarioId, req.user.id);
```

Comment:
can we actually have more than one lock per scenario? IIRC we have a unique constraint at db level (but it may make sense to keep the payload structure similar between projects and scenarios, so in any case I don't think that returning an array in `data` here is an issue).

Reply:
I think I misunderstood something on the MARXAN-1090 comment you posted:

> This could be queried in two ways:
>
>`GET /api/v1/projects/:projectId/editing-locks`
>
>`GET /api/v1/scenarios/:scenarioId/editing-locks`
>
>In the first case, we look up the list of scenarios in the project by the project id and return the lock status of them all.
>
>In the second case, we look up the parent project of whatever scenarioId we get, then likewise return the lock status >of all the scenarios in the project (so all the sibling scenarios of the one that happened to be used to query this data). >Of course we could instead only return the lock metadata of the given scenario, if a typical use case is to check only >the current lock status of the scenario the user is currently editing (so we can ignore changes to locks of sibling >scenarios). Query time will be negligible in all cases, but if users have hundreds of scenarios :exploding_head: then >we could save on response payload sizes here.

As you can see, the second case (scenario) should also return all the scenarios in the project and that's why I was returning an array as well here in `data`.

Reply:
lol, I had forgotten about my own evil plan. sorry.

so, great - functionally it's all good. Just see if you wish to make the evilness of the evil plan more explicit by renaming the functions involved in this "find on parent project by current scenario id".